tutor/scripts/test-server/backend.js

```diff
@@ -11,78 +11,91 @@ const os = require('os');
-  offerings: require('./backend/offerings'),
-  'courses/:courseId/dashboard': require('./backend/dashboard'),
-  'courses/:courseId/performance': require('./backend/performance'),
-  'courses/:courseId/guide': require('./backend/performance-forecast'),
-  'ecosystems/:ecosystemId/readings': require('./backend/readings'),
+    setrole: {
+        setRole() {  },
+        resetState() {  },
+        handler(req, resp) { setRole(req.query.role); resp.json({ ok: true }); },
+    },
+    resetState: {
+        setRole() {  },
+        resetState() {  },
+        handler(req, resp) { resetState(req.query.role); resp.json({ ok: true }); },
+    },
+    'user/bootstrap': require('./backend/bootstrap'),
+    offerings: require('./backend/offerings'),
+    'courses/:courseId/dashboard': require('./backend/dashboard'),
+    'courses/:courseId/performance': require('./backend/performance'),
+    'courses/:courseId/guide': require('./backend/performance-forecast'),
+    'ecosystems/:ecosystemId/readings': require('./backend/readings'),
 };
 
 const MULTI_HANDLERS = [
-  require('./backend/grading-templates'),
-  require('./backend/task-plans'),
-  require('./backend/courses'),
-  require('./backend/tasks'),
-  require('./backend/terms'),
-  require('./backend/exercises'),
-  require('./backend/course-roster'),
-  require('./backend/suggest'),
+    require('./backend/grading-templates'),
+    require('./backend/task-plans'),
+    require('./backend/courses'),
+    require('./backend/tasks'),
+    require('./backend/terms'),
+    require('./backend/exercises'),
+    require('./backend/course-roster'),
+    require('./backend/suggest'),
 ];
 
 // routes that have custom logic
 for (let route in GET_HANDLERS) {
-  server.get(`/api/${route}`, GET_HANDLERS[route].handler);
+    server.get(`/api/${route}`, GET_HANDLERS[route].handler);
 }
 MULTI_HANDLERS.forEach((handler) => {
-  handler.route(server);
+    handler.route(server);
 });
 
 server.use(jsonServer.rewriter({
-  '/api/user/ui_settings': '/ui-settings',
-  '/api/log/entry': '/log',
-  '/api/log/event/onboarding/:id': '/onboarding',
-  '/api/user/tours/:id': '/tours',
-  '/api/:key': '/:key',
-  '/api/courses/:courseId/plans*': '/previousTaskPlans',
+    '/api/user/ui_settings': '/ui-settings',
+    '/api/log/entry': '/log',
+    '/api/log/event/onboarding/:id': '/onboarding',
+    '/api/user/tours/:id': '/tours',
+    '/api/:key': '/:key',
+    '/api/courses/:courseId/plans*': '/previousTaskPlans',
 }));
 server.use(router);
 
 function setRole(role) {
-  for (let route in GET_HANDLERS) {
-    GET_HANDLERS[route].setRole(role);
-  }
-  MULTI_HANDLERS.forEach((handler) => { handler.setRole(role); });
+    for (let route in GET_HANDLERS) {
+        GET_HANDLERS[route].setRole(role);
+    }
+    MULTI_HANDLERS.forEach((handler) => { handler.setRole(role); });
+}
+function resetState() {
+    for (let route in GET_HANDLERS) {
+        GET_HANDLERS[route].resetState();
```

Comment:
should we check if each route has a `resetState`? That way we don't have to always write a `resetState` in each of the future routes.

Reply:
no, I don't think we should use this unless we absolutely have to.

Jest runs multiple processes concurrently, so if we happen to call setState while another process is in the middle of building an assignment it will reset the the state that the other spec had carefully setup.

Reply:
ug I'm sorry, I completely mis-read your comment @camliu89 - I took it to mean adding it to beforeEach, but you were thinking of checking for the fn on the handler before calling it.  that's not a bad idea, I'll update to do that